programs/sunrise-stake/src/lib.rs

```diff
@@ -352,6 +459,85 @@ pub struct Deposit<'info> {
     pub marinade_program: Program<'info, MarinadeFinance>,
 }
 
+#[derive(Accounts, Clone)]
+pub struct DepositStakeAccount<'info> {
+    #[account(has_one = marinade_state)]
+    pub state: Box<Account<'info, State>>,
+
+    #[account()]
+    pub marinade_state: Box<Account<'info, MarinadeState>>,
+
+    #[account(
+        mut,
+        constraint = gsol_mint.mint_authority == COption::Some(gsol_mint_authority.key()),
+    )]
+    pub gsol_mint: Box<Account<'info, Mint>>,
+
+    #[account(
+    seeds = [
+    state.key().as_ref(),
+    GSOL_MINT_AUTHORITY,
+    ],
+    bump = state.gsol_mint_authority_bump,
+    )]
+    pub gsol_mint_authority: SystemAccount<'info>,
```

Comment:
Should this be a system account rather than a PDA? I noticed that it was labelled a system account in the previous instruction so I followed that convention.

Reply:
As I understand it SystemAccount actually just means that the account has no data ie is owned by the system program, but the PDA is derived from the sunrise program. So I think this works.

Reply:
Got it. Any changes you would like me to make?

Reply:
No, I think we're good